roles/foreman_repositories/tasks/redhat_staging_repos.yml

```diff
@@ -6,39 +6,63 @@
   tags:
     - packages
 
-- name: 'Foreman {{ foreman_repositories_version }} Koji repository'
-  yum_repository:
-    name: foreman-koji
-    description: "Foreman {{ foreman_repositories_version }} Koji Repository"
-    baseurl: "http://koji.katello.org/releases/yum/foreman-{{ foreman_repositories_version }}/el{{ ansible_distribution_major_version }}/x86_64/"
-    priority: "1"
-    gpgcheck: no
-  tags:
-    - packages
+- when: foreman_repositories_staging_source == 'koji'
+  block:
```

Comment:
instead of a block, you could have dedicated task files for koji and stagingyum? feels cleaner

Reply:
Aight, take a look see